zcash_client_sqlite/src/lib.rs

```diff
@@ -1052,8 +1052,16 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
     ) -> Result<(), Self::Error> {
         self.transactionally(|wdb| {
             let tx_ref = wallet::put_tx_data(wdb.conn.0, d_tx.tx(), None, None)?;
+            let funding_accounts = wallet::get_funding_accounts(wdb.conn.0, d_tx.tx())?;
+            let funding_account = funding_accounts.iter().next().copied();
+            if funding_accounts.len() > 1 {
+                warn!(
+                    "More than one wallet account detected as funding transaction {:?}, selecting {:?}",
```

Comment:
I believe that this behaviour will lead to inconsistent views for the non-selected accounts, in exactly the way that @AArnott originally raised. It won't occur _often_ because multi-account funded transactions should be rare (as the approach is generally frowned upon), but it could happen. I think it's okay to choose this behaviour for now however, because any change we'd make to enable full funding-account tracking would require a database migration, at which point we'd just run `get_funding_accounts` again inside the migration.

Separately, do we need to pair this change with a migration to change how we recorded past notes? AFAICT the change in this PR will only affect future notes in the wallet, and the views will remain invalid for past notes. The data migration in that case would be the same as I mention above (except selecting a single account in the migration), so I think it's fine to omit this change in this PR (as newly recovered-from-seed wallets will be fine with just this change) and then add a migration later. If no migration is added to this PR, open an issue to do so.

Reply:
To be clear, since I don't fully follow @str4d's comment, I don't really care about transactions that are funded from multiple accounts, as to my knowledge no wallet enabled users to do that, so it's unlikely to come up in account recovery scenarios. 
What I _do_ care about is that in a multi-account wallet, transactions are correctly attributed as having been funded by the _one_ account that applies.

Reply:
Oh, also, I had already opened https://github.com/zcash/librustzcash/issues/1305 to track the remaining work here.